sycl/doc/extensions/proposed/sycl_ext_oneapi_reusable_events.asciidoc

```diff
@@ -0,0 +1,460 @@
+the `property::queue::enable_profiling` property or if the event `e` was created
+with the `ext::oneapi::experimental::enable_profiling` property.
+The event's `info::event_profiling::command_submit` timestamp reflects the time
+at which `enqueue_event_signal` is called.
+The event's `info::event_profiling::command_end` timestamp reflects the time at
+which the event enters the "complete" state.
+The event's `info::event_profiling::command_start` timestamp reflects the time
+that the tag operation starts executing.
+This timestamp is between the `info::event_profiling::command_submit` and
+`info::event_profiling::command_end` timestamps.
+
+It is unspecified whether the event ever has the
+`info::event_command_status::running` status.
+Implementations are encouraged to transition the event directly from the
+"submitted" status to the "complete" status and are encouraged to set the
+"command_start" timestamp to the same value as the "command_end" timestamp.
+
+[_Note:_ In order to understand why the "command_start" and "command_end"
+timestamps are encouraged to be the same, think of the tag operation as an empty
+kernel with an implicit set of dependencies on all previous commands in the
+same queue.
+This theoretical kernel starts executing when the dependencies are resolved.
+Since the kernel is empty, the end time is the same as the start time.
+The "command_start" and "command_end" timestamps are not required to be the
+same, though, in order to accommodate an implementation where the tag operation
+is implemented by submitting an actual kernel, which has non-zero execution
+time.
+_{endnote}_]
+
+_Throws:_ An `exception` with the `errc::invalid` error code if `evt` and `q`
+don't have the same context.
+
+=== Interaction with other event APIs
+
+An event _E_ created via `make_event` can be used as a command dependency (e.g.
+via `handler::depends_on`) for a command submitted to some queue _Q_.
+It is _not_ necessary for the context of _E_ to match the context of _Q_.
+
+If an event _E_ is used as a command dependency for some command _C_ (e.g. via
+`handler::depends_on`), the dependency is captured at the point when _C_ is
+submitted.
+It is legal to reassociated the event _E_ to a new command via
+`enqueue_event_signal` even before command _C_ completes.
+Doing so does _not_ change the dependency for command _C_.
+
+If another host thread is blocked waiting for event _E_ to complete via
+`event:wait` or `event::wait_and_throw` when event _E_ is reassociated with a
+new command via `enqueue_event_signal`, the behavior of the `event:wait` or
+`event::wait_and_throw` call is undefined.
+
+
+== Examples
+
+=== Recording cross queue dependencies
+
+[source,c++]
+----
+#include <sycl/sycl.hpp>
+namespace syclex = sycl::ext::oneapi::experimental;
+
+static constexpr size_t N = 1024;
+
+int main() {
+  sycl::queue q1{sycl::property::queue::in_order{}};
+  sycl::queue q2{sycl::property::queue::in_order{}};
+  sycl::event e = syclex::make_event();
```

Comment:
In this case, which context it will be taken to create the event? The default context of q1 or q2?